ignite/metrics/metric.py

```diff
@@ -296,7 +305,20 @@ def iteration_completed(self, engine: Engine) -> None:
                     f"it should contain {self.required_output_keys} keys, but given {list(output.keys())}"
                 )
             output = tuple(output[k] for k in self.required_output_keys)
-        self.update(output)
+
+        if isinstance(output, Sequence) and all([_is_list_of_tensors_or_numbers(o) for o in output]):
+            if not (len(output) == 2 and len(output[0]) == len(output[1])):
+                raise ValueError(
+                    f"Output should have 2 items of the same length, "
+                    f"got {len(output)} and {len(output[0])}, {len(output[1])}"
+                )
+            for o1, o2 in zip(output[0], output[1]):
+                # o1 and o2 are list of tensors or numbers
+                tensor_o1 = _to_batched_tensor(o1)
+                tensor_o2 = _to_batched_tensor(o2, device=tensor_o1.device)
```

Comment:
Should we move the output to a device or let this be done in the training step function ?

Reply:
here output is something coming from training step and in this case, o2 is a number that have to be wrapped with a tensor and put to appropriate device...

Reply:
My bad !!